sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -126,16 +125,13 @@ class JacksonParser(
 
         case VALUE_STRING =>
           // Special case handling for NaN and Infinity.
-          val value = parser.getText
-          val lowerCaseValue = value.toLowerCase(Locale.ROOT)
-          if (lowerCaseValue.equals("nan") ||
-            lowerCaseValue.equals("infinity") ||
-            lowerCaseValue.equals("-infinity") ||
-            lowerCaseValue.equals("inf") ||
-            lowerCaseValue.equals("-inf")) {
-            value.toFloat
-          } else {
-            throw new RuntimeException(s"Cannot parse $value as FloatType.")
+          parser.getText match {
+            case "NaN" | "+NaN" | "-NaN" => Float.NaN
```

Comment:
Yea, but it is supported in Scala. These were added to address https://github.com/apache/spark/pull/17956#discussion_r116163499 to say we support Scala  + Jackson standard.

```scala
scala> "+NaN".toDouble
res2: Double = NaN

scala> "-NaN".toDouble
res3: Double = NaN
```